logical/response.go

```diff
@@ -130,3 +131,23 @@ func ListResponseWithInfo(keys []string, keyInfo map[string]interface{}) *Respon
 
 	return resp
 }
+
+// RespondWithStatusCode takes a response and converts it to a raw response with
+// the provided Status Code.
+func RespondWithStatusCode(resp *Response, reqID string, code int) (*Response, error) {
```

Comment:
Rather than pass in the req ID, I suggest just passing in the request...same number of parameters, but a little more obvious.

Reply:
Done!